packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -439,6 +446,47 @@ export enum IpAddressType {
   DUAL_STACK = 'dualstack',
 }
 
+/**
+ * Configuration for a specific node type in OpenSearch domain
+ */
+export interface NodeConfig {
+  /**
+   * Whether this node type is enabled
+   *
+   * @default - false
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * The instance type for the nodes
+   *
+   * @default - m5.large.search
+   */
+  readonly type?: string;
+
+  /**
+   * The number of nodes of this type
+   *
+   * @default - 1
+   */
+  readonly count?: number;
+}
+
+/**
+ * Configuration for node options in OpenSearch domain
+ */
+export interface NodeOptions {
+  /**
+   * The type of node. Currently only 'coordinator' is supported.
+   */
+  readonly nodeType: 'coordinator';
```

Comment:
```suggestion
  readonly nodeType: NodeType;
```
Typing the property with an `enum` allows for extensibility in case new types are added in the future while avoiding breaking changes. `NodeType` needs to be declared of course.